docs/day-2-operations/Backups.md

```diff
@@ -0,0 +1,272 @@
+   oc rsh $(oc get pods -l 'deploymentConfig=backend-redis' -o json | jq '.items[0].metadata.name' -r) bash -c 'mv ${HOME}/data/dump.rdb ${HOME}/data/dump.rdb-old'
+   ```
+
+6. Rename the `appendonly.aof` file
+   ```
+   oc rsh $(oc get pods -l 'deploymentConfig=system-redis' -o json | jq '.items[0].metadata.name' -r) bash -c 'mv ${HOME}/data/appendonly.aof ${HOME}/data/appendonly.aof-old'
+   ```
+
+7. Move the Backup file to the POD
+
+   ```
+   oc cp ./backend-redis-dump.rdb $(oc get pods -l 'deploymentConfig=backend-redis' -o json | jq '.items[0].metadata.name' -r):/var/lib/redis/data/dump.rdb
+   ```
+
+8. Re-deploy `system-redis` to load the backup
+   ```
+   oc rollout latest dc/backend-redis
+   ```
+
+9. Edit the `redis-config` configmap
+   ```
+   oc edit configmap redis-config
+   ```
+
+10. Uncomment `SAVE` comands in the `redis-config` configmap
+   ```
+    save 900 1
+    save 300 10
+    save 60 10000
+   ```
+
+11. Set `appendonly` to yes in the `redis-config` configmap
+   ```
+   appendonly yes
+   ```
+
+12. Re-deploy `backend-redis` to reload the default configurations
+   ```
+   oc rollout latest dc/backend-redis
+   ```
+
+### `system-redis`
+
+1. Edit the `redis-config` configmap
+   ```
+   oc edit configmap redis-config
+   ```
+
+2. Comment `SAVE` comands in the `redis-config` configmap
+   ```
+    #save 900 1
+    #save 300 10
+    #save 60 10000
+   ```
+
+3. Set `appendonly` to no in the `redis-config` configmap
+   ```
+   appendonly no
+   ```
+
+4. Re-deploy `system-redis` to load the new configurations
+   ```
+   oc rollout latest dc/system-redis
+   ```
+
+5. Rename the `dump.rb` file
+   ```
+   oc rsh $(oc get pods -l 'deploymentConfig=system-redis' -o json | jq '.items[0].metadata.name' -r) bash -c 'mv ${HOME}/data/dump.rdb ${HOME}/data/dump.rdb-old'
+   ```
+
+6. Rename the `appendonly.aof` file
+   ```
+   oc rsh $(oc get pods -l 'deploymentConfig=system-redis' -o json | jq '.items[0].metadata.name' -r) bash -c 'mv ${HOME}/data/appendonly.aof ${HOME}/data/appendonly.aof-old'
+   ```
+
+7. Move the Backup file to the POD
+
+   ```
+   oc cp ./system-redis-dump.rdb $(oc get pods -l 'deploymentConfig=system-redis' -o json | jq '.items[0].metadata.name' -r):/var/lib/redis/data/dump.rdb
+   ```
+
+8. Re-deploy `system-redis` to load the backup
+   ```
+   oc rollout latest dc/system-redis
+   ```
+
+9. Edit the `redis-config` configmap
+   ```
+   oc edit configmap redis-config
+   ```
+
+10. Uncomment `SAVE` comands in the `redis-config` configmap
+   ```
+    save 900 1
+    save 300 10
+    save 60 10000
+   ```
+
+11. Set `appendonly` to yes in the `redis-config` configmap
+   ```
+   appendonly yes
+   ```
+
+12. Re-deploy `system-redis` to reload the default configurations
+   ```
+   oc rollout latest dc/system-redis
+   ```
+
+## Open Issues
+* what about System services and sphinx (index)?
+* How to handle backup/restore of job queues (of different types). They  can be lost or maybe done twice!
```

Comment:
I guess this is a question for @3scale/system @3scale/backend teams

Reply:
So for system: some jobs are only stored in system-redis but using event logging to replay them
For now there is no `replay` feature in `system` and not all the jobs are triggered by an event, so they cannot be replayed, but we could work on that.

However the `replay` will only work for events no older than 1 week see: https://github.com/3scale/porta/blob/ba76f595b24792f40292bf04e566087b3c75593f/app/workers/purge_old_events_worker.rb#L7
and
https://github.com/3scale/porta/blob/ba76f595b24792f40292bf04e566087b3c75593f/app/lib/event_store/event.rb#L59-L61

Reply:
@hallelujah any ideas on how could we solve this?

Reply:
For backend:

Restoring queues could be a problem because some jobs would be run twice.
Queues are supposed to be empty most of the time. Otherwise it means that reports are taking too long to be effective and rate-limit is not accurate. I think that for that reason losing the last few jobs that got enqueued might be better than restoring a backup with some old jobs that already got run.

Reply:
@davidor I have been looking for an option for generating RDB files for just one of the dbs, but looks like it is an issue that the Redis team are not looking forward to addressing (https://github.com/antirez/redis/issues/5369). 

TL;DR: It is impossible today to generate a backup file per DB in Redis. The recommendation is to have different instances of Redis for each database.

To fix this, we should have two different Redis for backend, which means an architecture change on the proposed templates.

@andrewdavidmackenzie WDYT?

Reply:
OK, recommendation and David's advice sounds sound.

Not clear what to do yet about System.....?

Reply:
@andrewdavidmackenzie, as Backend uses the same redis, we can't backup/restore just one of the DBs. So we can't follow David's advise (that's the issue). So what do you think we should do?

Reply:
@davidor There is any way to minimize the restoration of the queues in the current architecture?

Reply:
Job queues will be lost if the system redis was not backed up. And there is no way to restore most of them. Except from state synchronization between System and Backend.

A redeploy of the system-app pod will resync states of objects between system and backend through the post deploy pod https://github.com/3scale/porta/blob/8d75e618eff1f3050a9fc0f77ecdd576503ee0a7/lib/tasks/openshift.rake#L33

Reply:
Issue created to address the split of the backend databases: https://issues.jboss.org/browse/THREESCALE-2745